### PR TITLE
Improving debug log for plugin

### PR DESCRIPTION
**Pull request by priteshbandi**
Improving debug log for plugin

Signed-off-by: Pritesh Bandi <pritesb@amazon.com>

### CODE REVIEW

File: plugin/plugin.go [hunk 1]
```diff
@@ -152,29 +148,34 @@ func (p *CLIPlugin) GenerateEnvelope(ctx context.Context, req *proto.GenerateEnv
...
 func run(ctx context.Context, pluginName string, pluginPath string, req proto.Request, resp interface{}) error {
+	logger := log.GetLogger(ctx)
+
 	// serialize request
 	data, err := json.Marshal(req)
 	if err != nil {
 		return fmt.Errorf("%s: failed to marshal request object: %w", pluginName, err)
 	}
 
+	logger.Debugf("Plugin %s request: %+v", req.Command(), string(data))
```
Comment (JeyJeyGao) [inline plugin/plugin.go:170]: Some requests contain binary data and we may don't want to log them.

Comment (priteshbandi) [reply plugin/plugin.go]: Why not? Our spec says binary data would be base64 encoded.

File: plugin/plugin.go [hunk 2]
```diff
@@ -98,90 +96,90 @@ func (p *CLIPlugin) GetMetadata(ctx context.Context, req *proto.GetMetadataReque
...
 		if jsonErr != nil {
+			logger.Debugf("Plugin %s error response: %s", req.Command(), jsonErr)
 			return proto.RequestError{
 				Code: proto.ErrorCodeGeneric,
 				Err:  fmt.Errorf("response is not in JSON format. error: %v stderr: %v", err, stderr)}
 		}
 		return re
 	}
 
+	logger.Debugf("Plugin %s response: %+v", req.Command(), string(stdout))
```
Comment (shizhMSFT) [inline plugin/plugin.go:182]:
```suggestion
	logger.Debugf("Plugin %s response: %s", req.Command(), string(stdout))
```

File: plugin/plugin.go [hunk 3]
```diff
@@ -98,90 +96,90 @@ func (p *CLIPlugin) GetMetadata(ctx context.Context, req *proto.GetMetadataReque
...
 	stdout, stderr, err := executor.Output(ctx, pluginPath, req.Command(), data)
 	if err != nil {
+		logger.Debugf("Plugin %s returned error", req.Command())
+		if exitError, ok := err.(*exec.ExitError); ok {
+			logger.Debugf("Plugin %s exit code: %s", req.Command(), exitError.ExitCode())
+		}
 		var re proto.RequestError
 		jsonErr := json.Unmarshal(stderr, &re)
 		if jsonErr != nil {
+			logger.Debugf("Plugin %s error response: %s", req.Command(), jsonErr)
```
Comment (shizhMSFT) [inline plugin/plugin.go:174]:
```suggestion
			logger.Debugf("Plugin %s error response: %v", req.Command(), jsonErr)
```

File: plugin/plugin.go [hunk 4]
```diff
@@ -98,90 +96,90 @@ func (p *CLIPlugin) GetMetadata(ctx context.Context, req *proto.GetMetadataReque
...
 		return fmt.Errorf("%s: failed to marshal request object: %w", pluginName, err)
 	}
 
+	logger.Debugf("Plugin %s request: %s", req.Command(), string(data))
 	// execute request
 	stdout, stderr, err := executor.Output(ctx, pluginPath, req.Command(), data)
 	if err != nil {
+		logger.Debugf("Plugin %s returned error", req.Command())
+		if exitError, ok := err.(*exec.ExitError); ok {
+			logger.Debugf("Plugin %s exit code: %s", req.Command(), exitError.ExitCode())
```
Comment (shizhMSFT) [inline plugin/plugin.go:169]:
```suggestion
			logger.Debugf("Plugin %s exit code: %d", req.Command(), exitError.ExitCode())
```

File: plugin/plugin.go (hunk 3 above)
Comment (shizhMSFT) [reply plugin/plugin.go:174]: or may be `%w`.

File: plugin/plugin.go [hunk 5]
```diff
@@ -98,90 +96,90 @@ func (p *CLIPlugin) GetMetadata(ctx context.Context, req *proto.GetMetadataReque
...
-	logger.Debugf("Plugin describe-key request: %+v", req)
-	var resp proto.DescribeKeyResponse
 	if req.ContractVersion == "" {
 		req.ContractVersion = proto.ContractVersion
 	}
+
+	var resp proto.DescribeKeyResponse
 	err := run(ctx, p.name, p.path, req, &resp)
-	logger.Debugf("Plugin describe-key response: %+v", resp)
 	return &resp, err
```
Comment (shizhMSFT) [inline plugin/plugin.go:112]: Should we do
```go
var resp proto.DescribeKeyResponse
if err := run(...); err != nil {
    return nil, err
}
return &resp, nil
```
and for all other commands?

Comment (priteshbandi) [reply plugin/plugin.go:112]: IMO current format is more compact.

Comment (shizhMSFT) [reply plugin/plugin.go:112]: But not in a good code style and not memory efficient.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Pull request merged**